lib/v2/screens/profile_screens/security/security_screen.dart

```diff
@@ -67,14 +71,31 @@ class SecurityScreen extends StatelessWidget {
                         icon: const Icon(Icons.update),
                         title: 'Export Private Key'.i18n,
                         description: 'Export your private key so you can easily recover and access your account.'.i18n,
-                        onTap: () {},
+                        onTap: () => Share.share(settingsStorage.privateKey),
                       ),
-                      SecurityCard(
-                        icon: SvgPicture.asset('assets/images/key_guardians_icon.svg'),
-                        title: 'Key Guardians'.i18n,
-                        description:
-                            'Choose 3 - 5 friends and/or family members to help you recover your account in case.'.i18n,
-                        onTap: () {},
+                      BlocBuilder<SecurityBloc, SecurityState>(
+                        buildWhen: (previous, current) =>
+                            previous.hasNotification == false && current.hasNotification == true,
+                        builder: (context, state) {
+                          return Stack(
+                            children: [
+                              SecurityCard(
+                                icon: SvgPicture.asset('assets/images/key_guardians_icon.svg'),
+                                title: 'Key Guardians'.i18n,
+                                description:
+                                    'Choose 3 - 5 friends and/or family members to help you recover your account in case.'
+                                        .i18n,
+                                onTap: () {
+                                  if (state.hasNotification) {
```

Comment:
you should fire a tap event into the bloc

OnSecurityCardTapped()

Then the bloc can check in the state  if (state.hasNotification)

And do what it needs to do. 

Then the bloc sends a navigation command. 

You would have a BlocListener that listens to the commands, and act accordingly.

Reply:
Basically, move the logic into the bloc.